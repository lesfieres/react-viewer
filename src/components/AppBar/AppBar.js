/* eslint-disable prop-types */
import PropTypes from 'prop-types';
// eslint-disable-next-line no-unused-vars
import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Button from '@material-ui/core/Button';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';

const styles = () => ({
  appBar: {
    position: 'relative',
  },
  toolbarTitle: {
    flex: 1,
    textAlign: 'left',
  },
});

const RVAppBar = (props) => {
  const { classes } = props;

  return (
    <AppBar position="static" color="default" className={classes.appBar}>
      <Toolbar>
        <Typography
          variant="h6"
          color="inherit"
          noWrap
          className={classes.toolbarTitle}
        >
          Les fieres
        </Typography>
        <Button>Features</Button>
        <Button>Enterprise</Button>
        <Button>Support</Button>
        <Button color="primary" variant="outlined">
          Login
        </Button>
      </Toolbar>
    </AppBar>
  );
};

RVAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(RVAppBar);
