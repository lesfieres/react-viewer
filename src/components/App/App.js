/* eslint-disable prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import CssBaseline from '@material-ui/core/CssBaseline';
import { withStyles } from '@material-ui/core/styles';

import AppBar from '../AppBar/AppBar';
import logo from '../../logo.svg';
import './App.css';

const styles = theme => ({
  '@global': {
    body: {
      backgroundColor: theme.palette.common.white,
    },
  },
});

const App = (props) => {
  const { classes } = props;

  return (
    <React.Fragment>
      <CssBaseline />
      <div className="App">
        <header>
          <AppBar classes={classes} />
        </header>
        <div className="App-body">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/components/App/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </div>
      </div>
    </React.Fragment>
  );
};

App.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(App);
