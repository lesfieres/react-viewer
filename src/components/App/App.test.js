/* eslint-disable prop-types */
// eslint-disable-next-line no-unused-vars
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AppBar from '../AppBar/AppBar';
import App from './App';

jest.mock('../AppBar/AppBar');

describe('App component test', () => {
  beforeAll(() => {
    AppBar.mockImplementation(() => ({ render: () => <div /> }));
  });

  beforeEach(() => {
    AppBar.mockClear();
  });

  it('renders with mocked AppBar', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    expect(div.textContent.indexOf('Les fieres')).toEqual(-1);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
