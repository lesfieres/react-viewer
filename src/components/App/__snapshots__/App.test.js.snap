// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test matches snapshot 1`] = `
<div
  className="App"
>
  <header>
    <div />
  </header>
  <div
    className="App-body"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/components/App/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </div>
</div>
`;
